graylog2-server/src/main/java/org/graylog/plugins/views/search/searchtypes/export/ExportTabularResultResponse.java

```diff
@@ -77,7 +78,18 @@ public static AggregationWidgetExportResponse fromPivotResult(final PivotResult
                 })
                 .toList();
 
-        return new AggregationWidgetExportResponse(header, dataRows);
+        return new ExportTabularResultResponse(header, dataRows);
+    }
+
+    public static ExportTabularResultResponse fromMessageListResult(final MessageList.Result m) {
+        if(m.messages().isEmpty()) {
+            return new ExportTabularResultResponse(List.of(), List.of());
+        }
+
+        final var first = m.messages().get(0);
 
+        final var header = first.message().keySet().stream().toList();
+        final var rows = m.messages().stream().map(message -> new DataRow(header.stream().map(key -> message.message().get(key)).toList())).toList();
```

Comment:
It would be great to break this line into multiple lines for better readability.